:FC-899: :sparkles: creation of multiple lambda per events for Invocations by Skill (#12)
Templum/developers-community: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md

At the bottom of the page you can see a toggle that will activate/deactivate the second step, that allows you to configure a [whitelist](liveperson-functions-foundations-features.html#domain-whitelisting) entry.

#### Step: Lambda Skills

> To specify to which skill the lambda should react, simply select the skills during the creation/editing process of your lambda. By default, the lambda will react to all skills. If the lambda should only react to specific skills then there is a maximum of <b>50</b> skills that can be selected at any given time.

Author: updated

> Please note that lambda skills can only be assigned to controller bot events, for a overview of all the supported lambda skill events head over to our [Event Sources](liveperson-functions-event-sources-overview.html).

Author: updated

> Please note that lambda skills can only be assigned to controller bot events, for a overview of all the supported lambda skill events head over to our [Event Sources](liveperson-functions-event-sources-overview.html).

Author: resolved

You can create more than one lambda of the same event if you set different skills for each of the lambdas. If you do not assign a skill to the lambda you are only allowed to have one lambda of the given event. 
Some example scenarios:
✅ Event A, Skill 1,2 + Event A, Skill 3,4
✅ Event A, Skill 1,2 + Event A, Skill 3,4 + Event B, Default-skill
✅ Event A, Default-skill
❌ Event A, Skill 1,2 + Event A, Skill 3,4 +  Default-skill

> <img src="img/functions/faas-select-skill-2.png" alt="Functions select multiple skill" style="width:100%;"/>

Reviewer: Naming does not comply

Author: updated

> <div class="important">Changing the skills of a lambda does not require a redeployment. The change is reflected within 5 minutes after saving.</div>

Author: updated

At the bottom of the page you can see a toggle that will activate/deactivate the second step, that allows you to configure a [whitelist](liveperson-functions-foundations-features.html#domain-whitelisting) entry.

> ### Function Skills

Reviewer: No heading is needed here, it destroys the flow.

Suppose you would like to limit which agent should be allowed to invoke a function during an automated event trigger. In that case, function skills functionality will help set a function to a specific skill.

> Select the skills during your function's creation/editing process to specify which skill the function should react to; By default, the function will respond to all skills. If the function should only respond to specific skills, then there is a maximum of **50** skills that you can select at any given time.

Reviewer: ```suggestion
Skills can be selected via the drop-down menu, and it is populated using the skills available in the conversational cloud. By default, the function will respond to all skills. Please be aware that the maximum number of skills is **50**.

<img src="img/functions/faas_select_skill_2.png" alt="Functions select multiple skill" style="width:100%;"/>
```

Reviewer: You might also add a short sentence explaining how the drop-down looks/behaves for other events.

Reviewer: Further, this is **only** about the creation process, please explain the updating in the other section. If needed you can add a short section, where you also mention that updates may take up to 5 minutes and are not requiring a redeployment.

Reviewer: But updating is nothing that should be discussed here.

> Please note that you can only assign function skills to controller bot events; for an overview of all the supported functions skill events, head over to our [Event Sources](liveperson-functions-event-sources-overview.html#automatic-messages).

Reviewer: Covered in the introduction paragraph. It makes no sense to explain to people how to achieve and then provide them with limits. 

```suggestion
```

> You can create more than one function of the same event if you set different skills for each function. If you do not assign a skill to the function, you can only have one function of the given event. Some example scenarios:

Reviewer: ```suggestion
You can create more than one function of the same event if you set different skills for each function. Generally, as long as they are not overlapping, you can create multiple functions for the same event. The following Scenario should highlight the implications:
```

> ✅ Event A, Skill 1,2 + Event A, Skill 3,4

Reviewer: ```suggestion
✅  Function 1 for Event A [Skill 1,2] + Function 2 for Event A [Skill 3,4]
```

> ✅ Event A, Skill 1,2 + Event A, Skill 3,4 + Event B, Default-skill

Reviewer: Please follow the above naming to make it more clear, as people might not get that different functions are defined between the "+"

✅ Event A, Default-skill
❌ Event A, Skill 1,2 + Event A, Skill 3,4 + Default-skill

<img src="img/functions/faas_select_skill_2.png" alt="Functions select multiple skill" style="width:100%;"/>

{: .important}

> Changing the skills of a function does not require a redeployment. The change is reflected within 5 minutes after saving.

Reviewer: ```suggestion
The above example, "Default-skill", refers to not selecting any skill, the default behaviour.
```

Reviewer: The existing content is related to the update process, not the creation process.

At the bottom of the page you can see a toggle that will activate/deactivate the second step, that allows you to configure a [whitelist](liveperson-functions-foundations-features.html#domain-whitelisting) entry.

Suppose you want to ensure that a function is triggered only for specific skills as the logic is not required for every skill. We provide you with the ability to do so. Please note that this functionality is **limited** to the [Automatic Messages](liveperson-functions-event-sources-overview.html#automatic-messages). 

Skills can be selected via the drop-down menu, and it is populated using the skills available in the conversational cloud. By default, the function will respond to all skills. Please be aware that the maximum number of skills is **50**.

> 
> 

Reviewer: ```suggestion
```

You can create more than one function of the same event if you set different skills for each function. Generally, as long as they are not overlapping, you can create multiple functions for the same event. The following Scenario should highlight the implications:

✅  Function 1 for Event A [Skill 1,2] + Function 2 for Event A [Skill 3,4]
✅  Function 1 for Event A [Skill 1,2] + Function 2 for Event A [Skill 3,4] + Function 3 for Event B [Default-skill]
✅  Function 1 for Event A [Default-skill]
❌  Function 1 for Event A [Skill 1,2] + Function 2 for Event A [Skill 3,4] + [Default-skill]

> <img src="img/functions/faas_select_skill.png" alt="Functions select multiple skill" style="width:100%;"/>

Reviewer: The image should be above your edge case explanation. As it is not relevant to the picture.

further the image name should start with functions_*****.png like every other file

Author: updated

{: .notice}
The drop-down menu is always disabled if the selected function template is not an Automatic Messages template.

> 

Reviewer: This should be actually right before or after the explanation on how to set

Author: updated

To Update the skills on an existing function, you will find the skills drop-down menu inside the General information accordion menu on the right side of code Editor. Updating of skills does not require redeployment, the change is reflected within 5 minutes after saving.

> <img src="img/functions/faas_select_skill_edit.png" alt="Functions select multiple skill" style="width:100%;"/>

Author: I am looking at the file name and they are starting the same way mine are functions_**.png for example. functions_ui_debugger_sidebar.png, functions_ui_debugger_output.png

> <img src="img/functions/faas_select_skill_edit.png" alt="Functions select multiple skill" style="width:100%;"/>

Author: updated

{: .important}

> The above example, "Default-skill", refers to not selecting any skill, the default behaviour.

Author: moved
